cli/internal/fs/path.go

```diff
@@ -148,3 +149,39 @@ func ReadFile(fs afero.Fs, filename AbsolutePath) ([]byte, error) {
 func RemoveFile(fs afero.Fs, filename AbsolutePath) error {
 	return fs.Remove(filename.asString())
 }
+
+type pathValue struct {
+	base     AbsolutePath
+	current  *AbsolutePath
+	defValue string
+}
+
+func (pv *pathValue) String() string {
+	if *pv.current == "" {
+		return ResolveUnknownPath(pv.base, pv.defValue).ToString()
```

Comment:
IIUC, the issue is that if a user supplies `C:/unix/like/path`, we will pass it through as an `AbsolutePath`? I'm a little tempted to say "don't do that". Passing `/unix/like/path` on windows ends up getting treated as a relative path and the slashes appropriately handled. It is likely a user error, but not really one we can correct because we don't know the drive letter.

Reply:
I'd mostly just pitch running it through `Clean` afterward so that it's consistent.

Reply:
I did some research, and it appears that `/` is forbidden in windows filenames, so running `Clean` should be safe, as we are not removing their ability to reference a directory called "unix/like/path" contained within `C:`.

However, and I don't know if it matters, but *without* running it through `Clean`, some combination of the Go and windows filesystem APIs end up treating `/` as a separator. Calling `os.MkdirAll("C:/some", ...)` and `ioutil.WriteFile("C:/some/path", ...)` on windows successfully results in a file at `C:\some\path`.

I think I would've expected it to fail, given that `/` is both not the separator and not allowed in filenames.